src/Symfony/Component/HttpKernel/Attribute/ValueResolver.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class ValueResolver
```

Comment:
This attribute must be final or the retrieval should use the `INSTANCEOF` mode to support child classes (I suggest making it final in that case, as I see no reason to create child classes of that attribute that would force a specific name)

Reply:
So that’s what `ArgumentMetadata::IS_INSTANCEOF` is for!